master/buildbot/util/service.py

```diff
@@ -189,19 +191,39 @@ def getConfigDict(self):
                 'args': self._config_args,
                 'kwargs': self._config_kwargs}
 
+    @defer.inlineCallbacks
     def reconfigServiceWithSibling(self, sibling):
         # only reconfigure if sibling is configured differently.
         # sibling == self is using ComparableMixin's implementation
         # only compare compare_attrs
         if self.configured and sibling == self:
-            return defer.succeed(None)
+            defer.returnValue(None)
         self.configured = True
-        return self.reconfigService(*sibling._config_args,
-                                    **sibling._config_kwargs)
+        # render renderables in parallel
+        # Properties import to resolve cyclic import issue
+        from buildbot.process.properties import Properties
+        p = Properties()
+        p.master = self.master
+        # render renderables in parallel
+        renderables = []
+        kwargs = {}
+        accumulateClassList(self.__class__, 'renderables', renderables)
+        for k, v in sibling._config_kwargs.items():
+            if k in renderables:
+                value = yield p.render(v)
+                setattr(self, k, value)
+                kwargs.update({k: value})
+            else:
+                kwargs.update({k: v})
+        d = yield self.reconfigService(*sibling._config_args,
+                                       **sibling._config_kwargs)
+        defer.returnValue(d)
 
+    @defer.inlineCallbacks
     def configureService(self):
         # reconfigServiceWithSibling with self, means first configuration
-        return self.reconfigServiceWithSibling(self)
+        d = yield self.reconfigServiceWithSibling(self)
```

Comment:
I would prefer to avoid transforming this into inlineCallbacks since it is not needed. It adds an overhead